arenadata/arenadata_helper.go

```diff
@@ -40,3 +31,32 @@ func EnsureAdVersionCompatibility(backupVersion string, restoreVersion string) {
 func GetOriginalVersion(fullVersion string) string {
 	return adPattern.ReplaceAllString(fullVersion, "")
 }
+
+func PatchStatisticsStatements(backupConfig *history.BackupConfig, connectionPool *dbconn.DBConn, statements []toc.StatementWithType) []toc.StatementWithType {
+	// Backups created in versions 1.30.5_arenadata16 to 1.30.5_arenadata19 have ineffective sql for
+	// deleting statistics, which can affect restore performance. as a workaround for these
+	// versions, we enable nested loop.
+	if connectionPool.Version.Is("6") &&
+		strings.Contains(backupConfig.BackupVersion, "1.30.5_arenadata") {
+		arenadataVersion := getArenadataVersion(backupConfig.BackupVersion)
+
+		if arenadataVersion >= 16 && arenadataVersion <= 19 {
+			statements = append(statements, toc.StatementWithType{})
+			copy(statements[1:], statements[:])
+			statements[0] = toc.StatementWithType{
+				Statement: "SET enable_nestloop = ON;",
```

Comment:
Do we need `RESET enable_nestloop;` after finishing work?

Reply:
done